zapr.go

```diff
@@ -137,28 +186,32 @@ func (zl zapLogger) Enabled(lvl int) bool {
 
 func (zl *zapLogger) Info(lvl int, msg string, keysAndVals ...interface{}) {
 	if checkedEntry := zl.l.Check(toZapLevel(lvl), msg); checkedEntry != nil {
-		checkedEntry.Write(handleFields(zl.l, keysAndVals)...)
+		checkedEntry.Write(zl.handleFields(lvl, keysAndVals)...)
 	}
 }
 
 func (zl *zapLogger) Error(err error, msg string, keysAndVals ...interface{}) {
 	if checkedEntry := zl.l.Check(zap.ErrorLevel, msg); checkedEntry != nil {
-		checkedEntry.Write(handleFields(zl.l, keysAndVals, zap.Error(err))...)
+		checkedEntry.Write(zl.handleFields(noLevel, keysAndVals, zap.NamedError(zl.errorKey, err))...)
 	}
 }
 
 func (zl *zapLogger) WithValues(keysAndValues ...interface{}) logr.LogSink {
-	newLogger := zl.l.With(handleFields(zl.l, keysAndValues)...)
-	return newLoggerWithExtraSkip(newLogger, 0)
+	newLogger := *zl
```

Comment:
at this point, why not change the receiver to be a non-pointer?

Reply:
Unless there's a reason to use a pointer receiver, defaulting to a copy seems better - that's what I did in funcr and glogr

Reply:
I'm still not sure whether we are talking just about `WithValues` or all methods.

https://tour.golang.org/methods/8 says:
> In general, all methods on a given type should have either value or pointer receivers, but not a mixture of both. (We'll see why over the next few pages.)

The other receivers were pointers, so I also used a pointer here. I don't see a reason why pointers were used (no need to modify, struct is small), so I'll convert everything to copy.

Reply:
Actually, there is one reason for pointer: `Init` has to take a pointer.

We could keep that as pointer receiver and change the rest to value, but is that desirable? The FAQ advices against it (https://golang.org/doc/faq#methods_on_values_or_pointers). I don't think it matters here because in practice the only relevant method set is the one for `*zapLogger` which includes all methods, it's more about the basic principle.